src/embind/emval.js

```diff
@@ -195,22 +196,23 @@ var LibraryEmVal = {
 
     for (var i = 0; i < argCount; ++i) {
         functionBody +=
-            "var argType"+i+" = requireRegisteredType(Module['HEAP32'][(argTypes >>> 2) + "+i+"], \"parameter "+i+"\");\n" +
+            "var argType"+i+" = requireRegisteredType({{{ makeGetValue('argTypes', '0', '*') }}}, 'parameter "+i+"');\n" +
             "var arg"+i+" = argType"+i+".readValueFromPointer(args);\n" +
-            "args += argType"+i+"['argPackAdvance'];\n";
+            "args += argType"+i+"['argPackAdvance'];\n" +
+            "argTypes += {{{ POINTER_SIZE }}};\n";
     }
     functionBody +=
         "var obj = new constructor("+argsList+");\n" +
         "return valueToHandle(obj);\n" +
         "}\n";
 
     /*jshint evil:true*/
-    return (new Function("requireRegisteredType", "Module", "valueToHandle", functionBody))(
-        requireRegisteredType, Module, Emval.toHandle);
+    return (new Function("requireRegisteredType", "Module", "valueToHandle", "{{{ MEMORY64 ? "HEAPU64" : "HEAPU32" }}}" , functionBody))(
```

Comment:
This commit break my code:
1) we call for the first time with X Memory, everything OK
2) we call again for the 2nd time with Y > X memory , (with ALLOW_MEMORY_GROWTH)
it doesn't work and it returns `parameter 0 has unknown type emsc`

I had to remove from this line in my **.js** output the "HEAPU32" like was before or set HEAPU8 to make it work like v3.1.14

Reply:
Thanks!   Fix is in https://github.com/emscripten-core/emscripten/pull/17556

Reply:
Thank you very much I'm gonna test it ☺️
I opened an issue about , I m gonna close it if everything is fine